### PR TITLE
feat(OnLongPress): `options.onMouseUp` callback

### PR DESCRIPTION
**Pull request by noook**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

---

### Description

In issue #3203 it is briefly discussed that it's difficult to make `onLongPress` and `onClick` co-exist without having to hack it locally. This PR introduces the suggested logic into the `OnLongPress` renderless component.

### Additional context

- I wonder if we should add a way to use the click as an option to the composable
- I believe we can add this behaviour to the directive, but it relies on the composable and not the component, so finding a way to add this option to the composable could make sense
- This can be tested only if I add the click callback as an option to the composable
- I'll revert the commit for the demo, it is just here to show how it works
- I'm not sure whether I should update the documentation, as this is something that most users would do without expecting a different behaviour

### CODE REVIEW

**Comment by antfu** [comment]
I am not very sure about this, ideally the `click` event should be a native event and a component should not emit it. Meanwhile, if the logic only exists in the component version, the behavior misalignment with the composable one might end up confusing.

**Comment by noook** [comment]
I completely understand your point. I was trying to achieve what was described in the mentioned issue. Do you suggest doing it differently ? Like another event name so we keep the native click behavior, and possibly the `release` event ? all of that through the options ?

**Comment by antfu** [comment]
Maybe we could have an extra option to pass a handler, like `onMouseUp(duration: number, distance: number, isLongPress: boolean)` so users could pass custom logic to handle if they expect to trigger a click event or so?

File: packages/core/onLongPress/index.ts [hunk 1]
```diff
@@ -42,13 +50,41 @@ export function onLongPress(
...
+    startTimestamp = undefined
+    hasLongPressed = false
+  }
+
+  function onRelease(ev: PointerEvent) {
+    if (options?.modifiers?.self && ev.target !== elementRef.value)
+      return
+
+    if (!options?.onMouseUp)
+      return
```
Comment (noook) [inline packages/core/onLongPress/index.ts:71]: todo: early returns must also call `clear()`, or we could call clear early, and save a copy of variables that should be used in `options.onMouseUp()`

**Comment by noook** [comment]
Sorry messed up the rebase, will probably reopen the PR

**Review by antfu — APPROVED** [review]

**Pull request merged**